src/renderer/components/pages/CalendarPage.tsx

```diff
@@ -42,6 +43,45 @@ const CalendarPage = () => {
     setErrorMailboxes(mailboxes);
   };
 
+  // output type of userInfo
+  // judge whether userInfo is undefined
+  const isNoAccount = (userInfo: userInfoType | undefined): boolean => {
+    if (userInfo === undefined) {
+      console.log("userInfo is undefined");
+      return true;
+    }
+    if (userInfo.useraccounts.length === 0) {
+      console.log("userInfo.useraccounts is empty");
+      console.log(userInfo)
+      return true;
+    }
+    return false;
+  }
+
+  useEffect(() => {
```

Comment:
This is duplicated because the effect is already set in the SideBar component. Maybe it is better to keep the one here and remove the one in SideBar so that the props of SIdeBar can be simpler.